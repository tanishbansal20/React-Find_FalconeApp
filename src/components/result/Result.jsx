import React, {Component} from 'react';
import PropTypes from 'prop-types';
import './Result.scss';

export default class Result extends Component {
  static propTypes = {
    count: PropTypes.number.isRequired,
    planetName: PropTypes.string.isRequired,
  }

  render() {
    return (
        <div className="result">
          <h1>Finding Falcone!</h1>
          <p>Success! Congratulations for finding Falcone. King Shah is mighty pleased.</p>
          <p>Time Taken: {this.props.count}</p>
          <p>Planet Found: {this.props.planetName}</p>
          <button onClick={() => window.location.reload()}>Start Again</button>
        </div>
    );
  }
}
