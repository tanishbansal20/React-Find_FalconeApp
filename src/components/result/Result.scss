.result{
  text-align: center;
  padding: 20px;

  h1{
    margin-bottom: 80px;
  }

  p{
    font-size: 18px;
    font-weight: 600;
    margin-bottom: 25px;
  }

  button{
    background-color: #4CAF50;
    border: 3px solid black;
    color: black;
    padding: 15px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
  }
}
