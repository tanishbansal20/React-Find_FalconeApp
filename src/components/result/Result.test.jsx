import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import Result from "./Result.jsx";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders successfully with props", () => {
  act(() => {
    render(<Result count={200} planetName="Earth" />, container);
  });
  expect(container).toMatchSnapshot();
});
