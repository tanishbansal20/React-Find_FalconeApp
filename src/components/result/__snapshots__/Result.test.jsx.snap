// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully with props 1`] = `
<div>
  <div
    class="result"
  >
    <h1>
      Finding Falcone!
    </h1>
    <p>
      Success! Congratulations for finding Falcone. King Shah is mighty pleased.
    </p>
    <p>
      Time Taken: 
      200
    </p>
    <p>
      Planet Found: 
      Earth
    </p>
    <button>
      Start Again
    </button>
  </div>
</div>
`;
